src/systems/buoyancy_engine/BuoyancyEngine.cc

```diff
@@ -77,8 +80,32 @@ class ignition::gazebo::systems::BuoyancyEnginePrivateData
 
   /// \brief mutex for protecting bladder volume and set point.
   public: std::mutex mtx;
+
+  /// \brief get fluid density
+  public: double CurrentFluidDensity(
+    EntityComponentManager &_ecm,
```

Comment:
I think this can be const

```suggestion
    const EntityComponentManager &_ecm,
```